xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -4903,7 +4903,6 @@ glusterd_import_friend_volume(dict_t *peer_data, int count,
         (void)glusterd_volinfo_copy_brickinfo(old_volinfo, new_volinfo);
 
         (void)glusterd_delete_stale_volume(old_volinfo, new_volinfo);
-        glusterd_volinfo_unref(old_volinfo);
```

Comment:
I don' think we can delete volinfo unref  the volume will not delete ever because it will keep refcnt nonzero. 
This is a legacy code. The code was introduced by the patch (http://review.gluster.org/12329). Do you have a reproducer to reproduce the same?